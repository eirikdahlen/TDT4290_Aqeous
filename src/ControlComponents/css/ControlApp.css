.ControlApp {
  font-family: 'Segoe UI', sans-serif;
  height: 100%;
}

.controlFlex {
  height: 100%;
}

.topWindow {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  height: 60%;
}

/* Mode boxes */
.topWindow > div {
  width: 30%;
  height: 97%;
}

.bottomWindow {
  height: 40%;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  justify-content: space-between;
  width: 100%;
}

.bottomLeft {
  width: 50%;
  height: 100%;
}

.bottomRight {
  width: 50%;
  height: 100%;
  background-color: lightgray;
}

.lockFlex {
  display: flex;
  flex-direction: row;
  width: 60%;
}

.forcesTitle {
  font-size: 15px;
  font-weight: bold;
  padding: 5px 0px 5px 0px;
  text-align: center;
}
