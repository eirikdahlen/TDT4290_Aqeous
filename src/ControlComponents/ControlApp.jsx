import React, { useState, useEffect } from 'react';
import Values from './Values';
import ManualMode from './ManualMode';
import NetfollowingMode from './NetFollowingMode';
import DynamicPositioningMode from './DynamicPositioningMode';
import './css/ControlApp.css';

const { remote } = window.require('electron');

// The ControlApp is the main component for the controls-window
function ControlApp() {
  // Set the states to the global variables
  const [sensorValues, sensorUpdate] = useState(remote.getGlobal('fromROV'));
  const [controlValues, controlUpdate] = useState(remote.getGlobal('toROV'));
  const [mode, setMode] = useState(remote.getGlobal('mode'));
  const [IMCActive, setIMCActive] = useState(
    remote.getGlobal('settings').messageProtocol === 'IMC',
  );
  const [toIMC, setToIMC] = useState(remote.getGlobal('toROVIMC'));
  const [fromIMC, setFromIMC] = useState(remote.getGlobal('fromROVIMC'));
  const [settings, setSettings] = useState(remote.getGlobal('settings'));
  const [NFValues, setNFValues] = useState(remote.getGlobal('netfollowing'));
  const [DPValues, setDPValues] = useState(
    remote.getGlobal('dynamicpositioning'),
  );

  // make windows listen to ipc-msgs
  useEffect(() => {
    let interval = setInterval(() => {
      setToIMC(remote.getGlobal('toROVIMC'));
      setFromIMC(remote.getGlobal('fromROVIMC'));
      controlUpdate(remote.getGlobal('toROV'));
      sensorUpdate(remote.getGlobal('fromROV'));
      setMode(remote.getGlobal('mode'));
      setNFValues(remote.getGlobal('netfollowing'));
      setDPValues(remote.getGlobal('dynamicpositioning'));
    }, 300);
    window.ipcRenderer.on('settings-updated', () => {
      const currentSettings = remote.getGlobal('settings');
      setIMCActive(currentSettings.messageProtocol === 'IMC');
      setSettings({ ...currentSettings });
    });
    return () => {
      clearInterval(interval);
    };
  }, []);

  return (
    <div className="ControlApp">
      <div className="controlFlex">
        <div className="topWindow">
          <ManualMode
            title="Manual Mode"
            toROV={controlValues}
            modeData={mode}
          ></ManualMode>
          <NetfollowingMode
            title="Net Following"
            modeData={mode}
            step={0.05}
            values={NFValues}
          ></NetfollowingMode>
          <DynamicPositioningMode
            title="Dynamic Positioning"
            modeData={mode}
            step={0.1}
            fromROV={sensorValues}
            values={DPValues}
          ></DynamicPositioningMode>
        </div>
        <div className="bottomWindow">
          <div className="bottomLeft">
            <Values
              title="MESSAGES RECEIVED"
              values={IMCActive ? fromIMC : sensorValues}
              changeEffect={false}
              IMCActive={IMCActive}
            />
          </div>
          <div className="bottomRight">
            <Values
              IMCActive={IMCActive}
              title="MESSAGES SENT"
              values={IMCActive ? toIMC : controlValues}
              changeEffect={true}
              settings={settings}
            />
          </div>
        </div>
      </div>
    </div>
  );
}

export default ControlApp;
