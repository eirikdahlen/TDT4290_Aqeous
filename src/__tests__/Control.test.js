import React from 'react';
import renderer from 'react-test-renderer';
import ControlApp from './../ControlComponents/ControlApp';

test('renders correctly', () => {
  const tree = renderer.create(<ControlApp />).toJSON();
  expect(tree).toMatchSnapshot();
});
