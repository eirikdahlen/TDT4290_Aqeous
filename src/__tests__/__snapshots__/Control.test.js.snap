// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ControlApp"
>
  <h1>
    Hello Controls
  </h1>
</div>
`;
